src/coreclr/jit/earlyprop.cpp

```diff
@@ -524,7 +524,8 @@ GenTree* Compiler::optPropGetValueRec(unsigned lclNum, unsigned ssaNum, optPropK
 
         GenTree* treeRhs = ssaDefAsg->gtGetOp2();
 
-        if (treeRhs->OperIsScalarLocal() && lvaInSsa(treeRhs->AsLclVarCommon()->GetLclNum()))
+        if (treeRhs->OperIsScalarLocal() && lvaInSsa(treeRhs->AsLclVarCommon()->GetLclNum()) &&
+            treeRhs->AsLclVarCommon()->HasSsaName())
```

Comment:
Is it possible for `HasSsaName` to return `true` yet `lvaInSsa` be `false`? Seems counterintuitive.

Reply:
I agree, but yes, it's possible. For promoted structs that aren't tracked we sometimes give them the same SSA number as their tracked field if `LclVarDsc::CanBeReplacedWithItsField` returns true.

Reply:
That interaction is probably something to consider refactoring in .NET 7.0. If it is intended then I doubt the compiler is handling it correctly in all places.

Reply:
Agree. IIRC if it the condition I am thinking about, it is the fix for the regressions that came up after we stopped lying about struct types.

Reply:
I had to lookup IIRC:

**IIRC**
ABBREVIATION
IIRC (noun)if I recall correctly.
"IIRC, that was in the seventies"

Reply:
Looks like #41242 is related to this.